File: src/main/java/edu/ncrn/cornell/config/Ced2arConfig.java
package edu.ncrn.cornell.config;

import org.hibernate.validator.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

/**
 * @author Brandon Barker
 *
 * Created on 5/5/2016.
 */
@Configuration
@ConfigurationProperties(prefix = "ced2ar")
public class Ced2arConfig {
    @NotBlank
    private String uploadDir;
    //
    public void setUploadDir(String dir) {
        uploadDir = dir;
    }
    //
    public String getUploadDir() {
        return uploadDir;
    }




}

File: site/src/main/java/edu/ncrn/cornell/site/Ced2arConfig.scala
package edu.ncrn.cornell.site

import org.hibernate.validator.constraints.NotBlank
import org.springframework.boot.context.properties.{ConfigurationProperties, EnableConfigurationProperties}
import org.springframework.context.annotation.Configuration

import scala.beans.BeanProperty

/**
  * @author Brandon Barker
  *
  *         Created on 5/5/2016.
  */
@Configuration
//@EnableConfigurationProperties
@ConfigurationProperties(prefix = "ced2ar")
class Ced2arConfig {

  //TODO: how to make this boolean?
  @NotBlank
  @BeanProperty
  var checkDb: String = _

  @NotBlank
  @BeanProperty
  var uploadDir: String = _

  @NotBlank
  @BeanProperty
  var version: String = _


}